src/extlibs/hrvo/hrvo_agent.cpp

```diff
@@ -52,17 +52,24 @@ HRVOAgent::HRVOAgent(HRVOSimulator *simulator, const Vector &position,
       pref_speed_(max_speed_ * PREF_SPEED_SCALE),
       // TODO (#2676): This config should be dependency injected and updated when a
       // parameter is changed
-      obstacle_factory(TbotsProto::RobotNavigationObstacleConfig()),
-      ball_obstacle(std::nullopt)
+      obstacle_factory(TbotsProto::RobotNavigationObstacleConfig())
```

Comment:
if his is going to be overridden in the constructor, don't initialize it here

Reply:
The obstacle factory doesn't have a default constructor, so I'm forced to initialize it here first. I guess I can add a default constructor if you think that would be preferred